apps/web/src/pages/quick-start/components/demo-flow/nodes/NodeStep.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import { Popover } from '@mantine/core';
+
+import { useStyles } from '../../../../../design-system/template-button/TemplateButton.styles';
+import { colors, shadows, Text } from '../../../../../design-system';
+import { guidePreview, GuideTitleEnum, IBeat, IGuide } from './consts';
+import { useSegment } from '../../../../../components/providers/SegmentProvider';
+import { When } from '../../../../../components/utils/When';
+
+export function NodeStep({
+  data,
+  id,
+  Handlers,
+  Icon,
+  ActionItem,
+}: {
+  data: any;
+  id: string;
+  Handlers: React.FC<any>;
+  Icon: React.FC<any>;
+  ActionItem?: React.FC<any>;
+}) {
+  const { theme } = useStyles();
+  const { counter } = useCounter();
+  const [sequence, setSequence] = useState<IBeat>();
+
+  const label = data.label.toLowerCase();
+  const popoverData = guidePreview[label] as IGuide;
+  const titleGradient =
+    popoverData.title === GuideTitleEnum.DIGEST_PREVIEW || popoverData.title === GuideTitleEnum.DIGEST_PLAYGROUND
+      ? 'blue'
+      : 'red';
+
+  useEffect(() => {
+    setSequence(popoverData.sequence[counter.toString()] as IBeat);
+  }, [counter]);
```

Comment:
maybe we should introduce the `framer-motion` in the app? 
it looks easy if we would like to achieve that animation: https://codesandbox.io/s/uotor?module=/src/Example.tsx&file=/src/Example.tsx:584-588

wdyt? cc @scopsy

Reply:
I don't have previous experience with 'framer-motion but I would love to start using it in the project, however in this particular example when we use react-flow to render the nodes with their popovers I am not sure how to add the 
 framer `variants`.

Reply:
I don't have previous experience with 'framer-motion but I would love to start using it in the project, however in this particular example when we use react-flow to render the nodes with their popovers I am not sure how to add the framer `variants`.